components/brave_news/browser/background_history_querier.cc

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_news/browser/background_history_querier.h"
+
+#include <utility>
+
+#include "base/task/bind_post_task.h"
+#include "components/history/core/browser/history_types.h"
+
+namespace brave_news {
+
+BackgroundHistoryQuerier MakeHistoryQuerier(
+    base::WeakPtr<history::HistoryService> history_service,
+    base::RepeatingCallback<base::CancelableTaskTracker*()> get_tracker) {
+  auto history_sequence = base::SequencedTaskRunner::GetCurrentDefault();
+  return base::BindRepeating(
+      [](scoped_refptr<base::SequencedTaskRunner> history_sequence,
+         base::WeakPtr<history::HistoryService> history_service,
+         base::RepeatingCallback<base::CancelableTaskTracker*()> get_tracker,
+         QueryHistoryCallback callback) {
+        // |bound_callback| will always be invoked on the caller's thread.
+        auto bound_callback =
+            base::BindPostTaskToCurrentDefault(std::move(callback));
+
+        history_sequence->PostTask(
+            FROM_HERE,
+            base::BindOnce(
+                [](base::WeakPtr<history::HistoryService> service,
+                   base::RepeatingCallback<base::CancelableTaskTracker*()>
+                       get_tracker,
+                   QueryHistoryCallback callback) {
+                  auto* tracker = get_tracker.Run();
```

Comment:
And here 

`CHECK(!content::BrowserThread::CurrentlyOn(content::BrowserThread::UI);`